File: graphr/src/graphr/processing/VertexProcessingFacade.java
package graphr.processing;

import graphr.data.PrimData;
import graphr.graph.Vertex;

public interface VertexProcessingFacade {
	
	//-- Getting and setting local values
	
	public PrimData getValue(String key);
	public void setValue(String key, PrimData value);
	public long getId();

	//-- Agent replication
	
	public void broadcast();
	

}

File: graphrs/graphrs/processing/VertexProcessingFacade.swift
//
//  VertexProcessingFacade.swift
//  graphrs
//
//  Created by Marcus Sellmann on 25.03.16.
//  Copyright © 2016 CITBDA. All rights reserved.
//

protocol VertexProcessingFacade {
    //-- Getting and setting local values
    func getValue(key: String) -> PrimitiveData?
    func setValue(key: String, value: PrimitiveData)
    func getId() -> UInt64
    
    //-- Agent replication
    func broadcast()
}